### PR TITLE
feat: added url detection

### PR DESCRIPTION
**Pull request by Sathyanarayan09**

### CODE REVIEW

**Comment by viveksri96** [comment]
@Sathyanarayan09 Can you point me to the line number to review as this has?

File: src/redux/reducers/appDataReducers.js [hunk 1]
```diff
@@ -1,332 +1,365 @@
...
+  let { componentData } = state;
+  let { newState, id } = data;
+  componentData = componentData.map((component) => {
+    if (component.id === id) {
+      return {
+        ...component,
+        ...newState,
+        content: newState.content
+          ? component.componentType === "Text"
+            ? urlify(sanitizeHtml(newState.content))
```
Comment (Sathyanarayan09) [inline src/redux/reducers/appDataReducers.js:252]: @viveksri96 - added urlify here.

File: src/utils/helpers.js [hunk 2]
```diff
@@ -1,165 +1,194 @@
...
+    return `https://www.youtube.com/embed/${videoid}`;
+  }
+  return "";
+};
+
+export const urlify = (text) => {
+  var urlRegex = /(((https?:\/\/)|(www\.))[^\s]+)/g;
+  return text.replace(urlRegex, function (url, b, c) {
+    let url2 = c == "www." ? "http://" + url : url;
+    return '<a href="' + url2 + '" target="_blank">' + url + "</a>";
```
Comment (chandan-reddy-k) [inline src/utils/helpers.js:25]: @Sathyanarayan09 bro check this function once more know? I have a feeling this can be done differently

**Review by viveksri96 — APPROVED** [review]

**Comment by chandan-reddy-k** [comment]
@Sathyanarayan09 @viveksri96 this PR is not ready know? How was this approved and merged?

**Pull request merged**